### DISCUSSION TITLE
建议改名 运行环境 ➡️ 项目，创建运行环境➡️部署项目

### DISCUSSION THREAD

**Discussion by jlak0** [1Panel-dev/1Panel#5310] [Ideas]
### 1Panel 版本

v1.10.9-lts

### 请描述您的需求或者改进建议

个人理解这部分功能上应该更是应该称部署项目 deploy
环境基本上是docker容器内定了 基本没什么自定义
举个例子 我部署项目到Cloudflare，Vercel
CF和Vercel提供运行环境 我提供项目 
我通过Deploy让项目运行在CF提供的环境
1Panel里部署我的项目 我只选择/使用了运行环境
并没有进行创建环境

### 请描述你建议的实现方案

修改这名称后再引入新的创建运行环境
印象目前的创建运行环境类似docker compose简化ui
建议加入用户自选镜像 以方便创建python，deno，bun等项目

**Comment by zhengkunwang223** [comment]
感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。

**Comment by jlak0** [comment]
> 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。

确实在不同场景有不同的解释
我的理解下 1Panel并没有“创建”软件运行时需要的环境
而是“使用”了已经容器化好的运行环境/模板
如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板

**Comment by zhengkunwang223** [comment]
> > 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。
> 
> 确实在不同场景有不同的解释
> 我的理解下 1Panel并没有“创建”软件运行时需要的环境
> 而是“使用”了已经容器化好的运行环境/模板
> 如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板

创建只是一个操作，比如创建网站，创建容器等等，用户自己创建也可以叫创建，当然，1Panel的开放程度是很高的，不仅是代码开源，也可以建立自己的 PHP 运行环境，比如论坛 https://bbs.fit2cloud.com/t/topic/5201 这个帖子，就是新建了一个以 debian 为基础的 PHP-FPM 运行环境来满足特定软件的需求。

**Comment by jlak0** [comment]
> > > 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。
> > 
> > 确实在不同场景有不同的解释
> > 我的理解下 1Panel并没有“创建”软件运行时需要的环境
> > 而是“使用”了已经容器化好的运行环境/模板
> > 如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板
> 
> 创建只是一个操作，比如创建网站，创建容器等等，用户自己创建也可以叫创建，当然，1Panel的开放程度是很高的，不仅是代码开源，也可以建立自己的 PHP 运行环境，比如论坛 https://bbs.fit2cloud.com/t/topic/5201 这个帖子，就是新建了一个以 debian 为基础的 PHP-FPM 运行环境来满足特定软件的需求。

我不会php，有点不理解
node.js的环境创建自定义少
创建完好像也没什么可以复用
就像运行一行命令node server.js
或类似python main.js这么简单
所以才提了这个建议
仅代表个人理解 如果能作为参考最好了

**Comment by zhengkunwang223** [comment]
> > > > 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。
> > > 
> > > 确实在不同场景有不同的解释
> > > 我的理解下 1Panel并没有“创建”软件运行时需要的环境
> > > 而是“使用”了已经容器化好的运行环境/模板
> > > 如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板
> > 
> > 创建只是一个操作，比如创建网站，创建容器等等，用户自己创建也可以叫创建，当然，1Panel的开放程度是很高的，不仅是代码开源，也可以建立自己的 PHP 运行环境，比如论坛 https://bbs.fit2cloud.com/t/topic/5201 这个帖子，就是新建了一个以 debian 为基础的 PHP-FPM 运行环境来满足特定软件的需求。
> 
> 我不会php，有点不理解
> node.js的环境创建自定义少
> 创建完好像也没什么可以复用
> 就像运行一行命令node server.js
> 或类似python main.js这么简单
> 所以才提了这个建议
> 仅代表个人理解 如果能作为参考最好了

nodejs这个运行环境，需要有完整的项目代码才能运行，具体可以看我们 B 站视频

**Comment by jlak0** [comment]
> > > > > 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。
> > > > 
> > > > 确实在不同场景有不同的解释
> > > > 我的理解下 1Panel并没有“创建”软件运行时需要的环境
> > > > 而是“使用”了已经容器化好的运行环境/模板
> > > > 如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板
> > > 
> > > 创建只是一个操作，比如创建网站，创建容器等等，用户自己创建也可以叫创建，当然，1Panel的开放程度是很高的，不仅是代码开源，也可以建立自己的 PHP 运行环境，比如论坛 https://bbs.fit2cloud.com/t/topic/5201 这个帖子，就是新建了一个以 debian 为基础的 PHP-FPM 运行环境来满足特定软件的需求。
> > 
> > 我不会php，有点不理解
> > node.js的环境创建自定义少
> > 创建完好像也没什么可以复用
> > 就像运行一行命令node server.js
> > 或类似python main.js这么简单
> > 所以才提了这个建议
> > 仅代表个人理解 如果能作为参考最好了
> 
> nodejs这个运行环境，需要有完整的项目代码才能运行，具体可以看我们 B 站视频
>  

nodejs的运行环境我是用过的
我看了php的环境创建才懂了我们的理解的差异
php的确实符合创建运行环境这称呼，创建完也可复用
也不像nodejs需要马上绑定项目，1项目1环境

**Comment by zhengkunwang223** [comment]
> > > > > > 感谢建议，运行环境目前是为了提供一个含有编程语言的环境，在不同的场景中可能有不同的解释，在 1Panel 中可以理解为软件运行时需要的环境。
> > > > > 
> > > > > 确实在不同场景有不同的解释
> > > > > 我的理解下 1Panel并没有“创建”软件运行时需要的环境
> > > > > 而是“使用”了已经容器化好的运行环境/模板
> > > > > 如果我没弄错的话 用户并不能自己创建像python，deno，bun这样的软件运行时需要的环境，而只能选择现有的node环境/模板
> > > > 
> > > > 创建只是一个操作，比如创建网站，创建容器等等，用户自己创建也可以叫创建，当然，1Panel的开放程度是很高的，不仅是代码开源，也可以建立自己的 PHP 运行环境，比如论坛 https://bbs.fit2cloud.com/t/topic/5201 这个帖子，就是新建了一个以 debian 为基础的 PHP-FPM 运行环境来满足特定软件的需求。
> > > 
> > > 我不会php，有点不理解
> > > node.js的环境创建自定义少
> > > 创建完好像也没什么可以复用
> > > 就像运行一行命令node server.js
> > > 或类似python main.js这么简单
> > > 所以才提了这个建议
> > > 仅代表个人理解 如果能作为参考最好了
> > 
> > nodejs这个运行环境，需要有完整的项目代码才能运行，具体可以看我们 B 站视频
> >  
> 
> nodejs的运行环境我是用过的
> 我看了php的环境创建才懂了我们的理解的差异
> php的确实符合创建运行环境这称呼，创建完也可复用
> 也不像nodejs需要马上绑定项目，1项目1环境

所以其实运行环境只是一个称呼，无需纠结。基于容器单进程的情况，我们没有做在一个 nodejs环境运行多个项目，如果有什么建议，可以随时反馈。
